lib/Dialect/RTL/RTLOps.cpp

```diff
@@ -493,6 +519,62 @@ static void print(OpAsmPrinter &p, RTLModuleOp op) {
                   /*printBlockTerminators=*/true);
 }
 
+/// Lookup the generator for the symbol.  This returns null on
+/// invalid IR.
+Operation *RTLModuleGeneratedOp::getGeneratorKindOp() {
+  auto topLevelModuleOp = (*this)->getParentOfType<ModuleOp>();
+  if (!topLevelModuleOp)
+    return nullptr;
+
+  return topLevelModuleOp.lookupSymbol(generatorKind());
+}
+
+static LogicalResult verifyRTLModuleGeneratedOp(RTLModuleGeneratedOp op) {
+  // Check that this instance is inside a module.
+  auto module = dyn_cast<mlir::ModuleOp>(op->getParentOp());
+  if (!module) {
+    op.emitOpError("should be embedded in a top-level module");
+    return failure();
+  }
+
+  auto referencedKind = op.getGeneratorKindOp();
+  if (referencedKind == nullptr)
+    return op.emitError("Cannot find generator definition '")
+           << op.generatorKind() << "'";
```

Comment:
This seems to duplicate the verification in both methods? Is it sufficient to not do the verification check in `getGeneratorKindOp` and rely on verification to ensure that the `RTLModuleGeneratedOp` is in a `ModuleOp`?